src/main/java/com/parkhomovsky/bookstore/controller/CategoryController.java

```diff
@@ -0,0 +1,78 @@
+package com.parkhomovsky.bookstore.controller;
+
+import com.parkhomovsky.bookstore.dto.book.BookDtoWithoutCategoryIds;
+import com.parkhomovsky.bookstore.dto.category.CategoryDto;
+import com.parkhomovsky.bookstore.dto.category.CategoryRequestDto;
+import com.parkhomovsky.bookstore.service.BookService;
+import com.parkhomovsky.bookstore.service.CategoryService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping(value = "/categories")
+@Tag(name = "Categories management",
+        description = "Endpoints for managing categories")
+public class CategoryController {
+    private final CategoryService categoryService;
+    private final BookService bookService;
+
+    @PostMapping
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @ResponseStatus(HttpStatus.CREATED)
+    @Operation(summary = "Create new category",
+            description = "Create a new category and add it to the database")
+    public CategoryDto create(@RequestBody @Valid CategoryRequestDto categoryRequestDto) {
+        return categoryService.create(categoryRequestDto);
+    }
+
+    @GetMapping
+    @Operation(summary = "Retrieve all categories", description = "Get list of all categories")
+    public List<CategoryDto> getAllCategories() {
+        return categoryService.getAll();
+    }
+
+    @PutMapping("/{id}")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @Operation(summary = "Update category",
+            description = "Update category in database with provided id and data")
+    public CategoryDto update(@PathVariable Long id,
+                               @Valid @RequestBody CategoryRequestDto categoryRequestDto) {
```

Comment:
Please be consistent and choose which order is better valid first or requestBody